patsylearn/patsy_adaptor.py

```diff
@@ -117,8 +124,34 @@ def predict(self, data):
         data : dict-like (pandas dataframe)
             Input data. Column names need to match variables in formula.
         """
-        X = np.array(dmatrix(self.design_X_, data))
-        return self.estimator_.predict(X)
+        X = dmatrix(self.design_X_, data, return_type=self.return_type)
```

Comment:
Is that how patsy does it? Changing the return type based on what libraries are installed seems very fragile to me.